### DISCUSSION TITLE
Eclipse 2022-12 hangs at startup for a very long time, with runnable thread in org.eclipse.m2e.core.ui.internal.M2EUIPluginActivator

### DISCUSSION THREAD

**Discussion by davidmichaelkarr** [eclipse-m2e/m2e-core#1216] [General]
I have been struggling with trying to use Eclipse 2022-12 for quite a while now. When I start it up, I get the prompt to delete the obsolete M2E cache, which I always decline, as I'm not convinced yet that 2022-12 is viable.  In any case, Eclipse hangs before it can even dismiss the dialog.  Every time I look at the thread dump, the top runnable thread has "org.eclipse.m2e.core.ui.internal.M2EUIPluginActivator" in the stack.  It's possible this will eventually finish, but it's been running for at least 20 minutes now.  My laptop does not have any hardware resource issues. I have 32gb of ram, and I've allocated 3gb for the Eclipse process.
[threaddump.txt](https://github.com/eclipse-m2e/m2e-core/files/10483752/threaddump.txt)

**Comment by mmoser18** [comment]
Had a similar hang. Started it with - clean option. Took a while, too, but finally succeeded.

**Comment by mickaelistria** [comment]
It doesn't look like m2e is to blame here. I suspect IOConsolePartitioner would actually be the culprit here. Do you have a Console open and already showing content at startup?

**Reply by davidmichaelkarr** [reply]
I'm not sure what you mean by that.  You mean in the Console View in Eclipse? In any case, the recommendation of using "-clean" actually seems to have made a difference.  It takes a little longer for the window to appear, but I haven't seen it freeze anymore since I added that.

**Reply by mmoser18** [reply]
Once you succeeded with - clean you can remove it again. Normally it only needs to be provided once. If Eclipse only starts with it then there is indeed is a bigger issue.

**Comment by davidmichaelkarr** [comment]
Unfortunately, it was just an illusion. The hang is back, whether or not I run with "-clean" or not. I tried leaving it running for the last few hours, and it's still hung.
[threaddump.txt](https://github.com/eclipse-m2e/m2e-core/files/10506683/threaddump.txt)

**Reply by mickaelistria** [reply]
And m2e is not present in that last dump, comforting the idea that the issue is in the Eclipse Platform console view. Please open an issue with those details to https://github.com/eclipse-platform/eclipse.platform.debug/issues
